pkg/utils/validation/admissionplugins/admissionplugins.go

```diff
@@ -115,24 +118,39 @@ func IsAdmissionPluginSupported(plugin, version string) (bool, error) {
 type AdmissionPluginVersionRange struct {
 	versionutils.VersionRange
 
-	Forbidden bool
-	Required  bool
+	Forbidden           bool
+	Required            bool
+	AllowsConfiguration bool
 }
 
-func getAllForbiddenPlugins() []string {
+func getAllForbiddenPlugins(version string) ([]string, error) {
 	var allForbiddenPlugins []string
 	for name, vr := range admissionPluginsVersionRanges {
-		if vr.Forbidden {
+		var (
+			inRange bool
+			err     error
+		)
+		if inRange, err = vr.Contains(version); err != nil {
+			return nil, err
+		}
+
+		if inRange && vr.Forbidden {
 			allForbiddenPlugins = append(allForbiddenPlugins, name)
 		}
 	}
-	return allForbiddenPlugins
+	return allForbiddenPlugins, nil
 }
 
 // ValidateAdmissionPlugins validates the given Kubernetes admission plugins against the given Kubernetes version.
 func ValidateAdmissionPlugins(admissionPlugins []core.AdmissionPlugin, version string, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
+	forbiddenPlugins := []string{}
+	allForbiddenPlugins, pluginErr := getAllForbiddenPlugins(version)
+	if pluginErr != nil {
+		allErrs = append(allErrs, field.InternalError(fldPath, pluginErr))
```

Comment:
```suggestion
		allErrs = append(allErrs, field.InternalError(fldPath, pluginErr))
		return allErrs
```

Nit: The code can exit early here, if there is such a profound error.